snakemake_storage_plugin_s3/__init__.py

```
@@ -325,10 +314,38 @@ def retrieve_object(self):
+
+        try:
+            self.s3obj().load()
+        except botocore.exceptions.ClientError as e:
+            err_code = e.response["Error"]["Code"]
+            if err_code == "400":
+                raise WorkflowError(
+                    f"Bad request for S3 object {self.query}. This can happen if "
+                    "the query contains unsupported characters (e.g. '..'), "
+                    "if your credentials are invalid, or if your permissions are "
+                    "insufficient."
+                ) from e
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)